builtin/logical/transit/path_encrypt.go

```diff
@@ -393,7 +426,19 @@ func (b *backend) pathEncryptWrite(ctx context.Context, req *logical.Request, d
 			warnAboutNonceUsage = true
 		}
 
-		ciphertext, err := p.Encrypt(item.KeyVersion, item.DecodedContext, item.DecodedNonce, item.Plaintext)
+		var factory interface{}
+
+		if item.AssociatedData != "" {
+			isAEAD := p.Type.String() == "aes128-gcm96" || p.Type.String() == "aes192-gcm96" || p.Type.String() == "aes256-gcm96" || p.Type.String() == "chacha20-poly1305"
```

Comment:
Could we switch this to use the existing pattern of adding this test on the KeyType type in `policy.go` around line 146, instead of using string tests? For example we seem to have added an "impossible" key type value in this test of `aes192-gcm96`

```
func (kt KeyType) AEADSupported() bool {
	switch kt {
	case KeyType_AES128_GCM96, KeyType_AES256_GCM96, KeyType_ChaCha20_Poly1305:
		return true
	}
	return false
}
```

Reply:
Yeah, I added it intentionally so we don't fail if we later introduce AES-192... This does seem cleaner since the change should be localized to KeyType.